src/shared_state.rs

```diff
@@ -199,10 +199,16 @@ impl SharedState {
             shared_state: shared_state.clone(),
         };
 
-        let inbound_thread = thread::spawn(move || inbound.read_loop());
+        let inbound_thread = thread::Builder::new()
+            .name("nats_inbound_read_loop".to_string())
+            .spawn(move || inbound.read_loop())
+            .expect("threads should be spawnable");
```

Comment:
while this looks like it's adding a new panic, thread::spawn will panic in the same way by default. I figure that since we're about to move to the single-thread model, we don't need to break the API to avoid panicking, as this will only be in place for a short while.